scripts/generate_headers

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+PROJECT_DIR="$SCRIPT_DIR/../"
+
+cd $PROJECT_DIR
+
+rm -rf ./target/headers
+mkdir -p ./target/headers
+cp -r ./rust/noosphere/include ./target/headers/
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+	# macos linker fails to generate all exports without this flag
+	# https://github.com/subconsciousnetwork/noosphere/issues/473
+	CARGO_PROFILE_DEV_CODEGEN_UNITS=1 cargo run --verbose --package noosphere --example generate_header --features headers --locked
```

Comment:
@cdata this can be set via `[profile.dev]` for macos only, but don't believe we can target only the header gen example.

Reply:
What are the side-effects? Is it just a perf difference?

Reply:
prevents LTO maybe? https://doc.rust-lang.org/cargo/reference/profiles.html#codegen-units

Reply:
serendipitous timing https://nnethercote.github.io/2023/07/11/back-end-parallelism-in-the-rust-compiler.html

Reply:
Let's just stick to the script as written. We can revisit if it seems unsatisfying down the road, and I would rather not compromise all builds for the sake of this problem.

Reply:
Aye definitely not something we want on all builds, this just runs on macos's header-gen binary

Reply:
This gets the job done, but there's a side effect of needing to rebuild more often as this flag invalidates debug builds (that are missing this flag)